Remove UpdateTests' dependency on groovy.

jpountz opened the pull request:
These tests had to be moved to lang-groovy when groovy has been made a plugin.
I refactored them a bit to use mock plugins instead so that groovy is not
necessary anymore and they can come back to core.

jpountz commented:
Woops @javanna worked on AliasesRoutingTests in parallel. So you can just focus on UpdateTests for now.

nik9000 commented on core/src/test/java/org/elasticsearch/routing/AliasRoutingIT.java:
```diff
@@ -40,13 +48,111 @@
+
+    }
+
+    public static class MockScriptEngine implements ScriptEngineService {
```
I wonder if you could refactor this into an abstract superclass? I think I've seen this kind of thing scattered around the tests.

nik9000 replied:
And if you did that then MockScriptPlugin could also become a utility pretty easily I think.

rjernst replied:
I don't think it needs to be abstract, but I think you should inherit from `org.elasticsearch.script.MockScriptEngine`. Or refactor that to allow more custom logic (right now search/execute do nothing).

nik9000 replied:
Yeah - I don't care about abstract - just that it'd be nice if you would refactor this so it looks its more obvious what is special about each of the mock engines without having to read 20 lines of boiler plate each.

javanna commented on core/src/test/java/org/elasticsearch/routing/AliasRoutingIT.java:
```diff
@@ -77,7 +183,7 @@ public void testAliasCrudRouting() throws Exception {
         client().prepareUpdate("alias0", "type1", "1")
                 .setUpsert(XContentFactory.jsonBuilder().startObject().field("field", 1).endObject())
-                .setScript(new Script("ctx._source.field = 'value2'", ScriptService.ScriptType.INLINE, null, null))
+                .setScript(new Script("", ScriptService.ScriptType.INLINE, MockScriptEngine.NAME, Collections.singletonMap("field", "value2")))
```
Having worked on this yesterday, I did see that some of the tests will need a mock, but I think we can take out the scripting aspect from this one. The test aims at verifying that whatever update works against an alias. We can then provide a partial document instead of a script like I did in #13945?

javanna commented:
Left a comment on `AliasRoutingIT`. The update test looks great, thanks for taking the time to fix it.

jpountz commented:
I simplified a bit the script engine API in master (removing two methods) and rebased this PR so that less ceremony is required to implement a script engine. Is it good enough? I'm a bit on the fence to extend MockScriptEngine as it might encourage to add hacks to the base class when needed for some tests, so I tend to like keeping things separated. I suspect part of the issue is that our scripting API is a bit too complicated for what it does?

javanna commented:
I agree with you @jpountz I would merge as-is.
